src/main/java/com/suse/saltstack/netapi/parser/SaltStackParser.java

```diff
@@ -1,32 +1,39 @@
 package com.suse.saltstack.netapi.parser;
 
 import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
 import com.suse.saltstack.netapi.exception.SaltStackParsingException;
+import com.suse.saltstack.netapi.results.*;
 
 import java.io.BufferedReader;
 import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.io.Reader;
-import java.lang.reflect.Type;
+import java.util.List;
 
-/**
- * ABS for result parsers. All result parsers need to inherit from this class. Generic methods
- * that apply to multiple results can be added to this class.
- */
-public abstract class SaltStackParser<T> {
-    public abstract T parse(InputStream inputStream) throws SaltStackParsingException;
+
+public class SaltStackParser<T> {
```

Comment:
Please add javadoc comments at least to all classes as well as to all public methods.